api_server.py

```
@@ -74,38 +86,45 @@ def format_hours(minutes: Optional[int]) -> Optional[str]:
-            display += " tomorrow"
-        else:
-            display += end_time.strftime(" on %d/%m")
-
-        return {
-            "available": True,
-            "duration": format_hours(duration_min),
-            "end_time_display": display,
-        }
+        sql = (
+            f"SELECT end_time FROM {table} "
+            f"WHERE {col} = ? AND start_time <= ? AND end_time > ? LIMIT 1"
```

Ruff (0.15.1)

[error] Possible SQL injection vector through string-based query construction

(S608)